numpy/_core/arrayprint.py

```diff
@@ -1062,8 +1048,8 @@ def fillFormat(self, data):
         # if there are non-finite values, may need to increase pad_left
         if data.size != finite_vals.size:
             neginf = self.sign != '-' or any(data[isinf(data)] < 0)
-            nanlen = len(_format_options['nanstr'])
-            inflen = len(_format_options['infstr']) + neginf
+            nanlen = len(_format_options.get()['nanstr'])
```

Comment:
Given that these are used twice, in principle it would seem to make more sense to have `self.nanstr` and `self.infstr`, but that is admittedly a bit out of scope.